lib/sections/publish.js

```diff
@@ -1,79 +1,62 @@
 var utils = require(__dirname + '/utils.js');
 var uuid  = require('node-uuid');
 var os    = require('os');
-// var i18n  = require('i18n');
-// var time  = require('time');
 
 var publish = function(eventType, args, callback){
   var self = this;
   var queue = self.options.incommigQueue;
-  var klass = self.options.busDriverClassKey;
-
-  payload = [publishMetadata(eventType, args)];
+  var klass = self.options.busClassKey;
+  var payload = [publishMetadata(eventType, args)];
+  payload[0].bus_class_proxy  = '::QueueBus::Driver';
 
   self.queueObject.enqueue(queue, klass, payload, function(err, toRun){
     if(typeof callback === 'function'){ callback(err, toRun); }
   });
-}
+};
 
 var publishAt = function(timestamp, eventType, args, callback){
-  var self = this;
-  var queue = self.options.incommigQueue;
-  var klass = self.options.busPublisherClassKey;
-
-  payload = [publishMetadata(eventType, args)];
-  if(payload["bus_delayed_until"] == null){
-    payload["bus_delayed_until"] = Math.floor(timestamp/1000);
+  var self    = this;
+  var queue   = self.options.incommigQueue;
+  var klass   = self.options.busClassKey;
+  var payload = [publishMetadata(eventType, args)];
+  payload[0].bus_class_proxy  = '::QueueBus::Publisher';
+
+  if(payload.bus_delayed_until === undefined){
+    payload.bus_delayed_until = Math.floor(timestamp/1000);
   }
-  delete payload['bus_published_at']; // will get re-added upon re-publish
 
-  self.queueObject.enqueueAt(timestamp, queue, klass, payload, function(){
-    if(typeof callback === 'function'){ callback(); }
+  delete payload.bus_published_at; // will get re-added upon re-publish
+
+  self.queueObject.enqueueAt(timestamp, queue, klass, payload, function(err){
+    if(typeof callback === 'function'){ callback(err); }
   });
-}
+};
 
 var publishIn = function(time, eventType, args, callback){
   var self = this;
   var timestamp = new Date().getTime() + time;
   self.publishAt(timestamp, eventType, args, callback);
-}
+};
 
 var publishMetadata = function(eventType, args){
   var payload = {};
-  if(eventType != null){
-    payload['bus_event_type'] = eventType
+  if(eventType){
+    payload.bus_event_type = eventType;
   } else {
-    payload['bus_event_type'] = null;
+    payload.bus_event_type = null;
   }
-  payload['bus_created_at']   = utils.timestamp(); // TODO: get this back in ruby resque-bus
-  payload['bus_published_at'] = utils.timestamp();
-  payload['bus_id']           = utils.timestamp() + "-" + uuid.v4();
-  payload['bus_app_hostname'] = os.hostname();
-  // payload['bus_locale']       = i18n.getLocale();
-  // payload['bus_timezone']     = time.currentTimezone; 
+  payload.bus_created_at   = utils.timestamp(); // TODO: get this back in ruby resque-bus
+  payload.bus_published_at = utils.timestamp();
+  payload.bus_id           = utils.timestamp() + "-" + uuid.v4();
+  payload.bus_app_hostname = os.hostname();
```

Comment:
here are the fields
https://github.com/queue-bus/queue-bus/blob/f5822ce1da81fd4cae4191c59726cbcd7cbd702d/lib/queue_bus/publishing.rb#L24

bus_published_at
bus_event_type
bus_id
bus_app_hostname
bus_locale
bus_timezone